tiledb/ml/readers/pytorch_sparse.py

```diff
@@ -126,76 +164,87 @@ def __iter__(self) -> Iterator[DataType]:
+                x_buffer_csr = self.__to_csr("X", x_buffer, offset)
+
+                if isinstance(self.y, tiledb.SparseArray):
+                    y_buffer_csr = self.__to_csr("Y", y_buffer, offset)
+
                 # Split the buffer_size into batch_size chunks
-                # batch_offsets = np.arange(offset, min(offset + self.buffer_size, iter_end), self.batch_size)
                 batch_offsets = np.arange(0, self.buffer_size, self.batch_size)
 
                 # Shuffle offsets in case we need batch shuffling
                 if self.batch_shuffle:
                     np.random.shuffle(batch_offsets)
 
                 for batch_offset in batch_offsets:
-                    x_batch = {
-                        attr: data[batch_offset : batch_offset + self.batch_size]
-                        for attr, data in x_buffer.items()
-                    }
-                    y_batch = {
-                        attr: data[batch_offset : batch_offset + self.batch_size]
-                        for attr, data in y_buffer.items()
-                    }
-
-                    x_coords = []
-                    for i in range(0, self.x.schema.domain.ndim):
-                        dim_name = self.x.schema.domain.dim(i).name
-                        x_coords.append(x_batch[dim_name])
-
-                    # Normalise indices for torch.sparse.Tensor We want the coords indices in every iteration
-                    # to be in the range of [0, self.batch_size] so the torch.sparse.Tensors can be created batch-wise.
-                    # If we do not normalise the sparse tensor is being created but with a dimension [0, max(coord_index)],
-                    # which is overkill
-                    x_coords[0] -= x_coords[0].min()
+                    x_batch = x_buffer_csr[
+                        batch_offset : batch_offset + self.batch_size
+                    ]
+
+                    if x_batch.data.size == 0:
+                        return
+
+                    if isinstance(self.y, tiledb.SparseArray):
+                        y_batch = y_buffer_csr[
+                            batch_offset : batch_offset + self.batch_size
+                        ]
+                    else:
+                        y_batch = {
+                            attr: data[batch_offset : batch_offset + self.batch_size]
+                            for attr, data in y_buffer.items()
+                        }
+
+                    # Keep row records number for cross-check between X and Y batches. Last index excluded shows to
+                    # empty
+                    samples_num_x = x_batch.indptr[:-1]
+
+                    # Transform back to COO for torch.sparse_coo_tensor to digest
+                    x_batch_coo = x_batch.tocoo()
+                    x_coords = np.stack(
+                        (x_batch_coo.row, x_batch_coo.col), axis=-1
+                    ).tolist()
```

Comment:
`tolist()` is probably redundant, `torch.tensor` can be initialized by [numpy arrays](https://pytorch.org/docs/stable/tensors.html)